AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -608,23 +663,41 @@ - (ASLayout *)__measureWithSizeRange:(ASSizeRange)constrainedSize
   ASDisplayNodeAssertTrue(_layout.size.width >= 0.0);
   ASDisplayNodeAssertTrue(_layout.size.height >= 0.0);
 
-  // we generate placeholders at measureWithSizeRange: time so that a node is guaranteed to have a placeholder ready to go
-  // also if a node has no size, it should not have a placeholder
-  if (self.placeholderEnabled && [self _displaysAsynchronously] && _layout.size.width > 0.0 && _layout.size.height > 0.0) {
-    if (!_placeholderImage) {
-      _placeholderImage = [self placeholderImage];
-    }
-
-    if (_placeholderLayer) {
-      [self setupPlaceholderLayerContents];
-    }
+  // we generate placeholders at measureWithSizeRange: time so that a node is guaranteed
+  // to have a placeholder ready to go. Also, if a node has no size it should not have a placeholder
+  if (self.placeholderEnabled && [self _displaysAsynchronously] &&
+      _layout.size.width > 0.0 && _layout.size.height > 0.0) {
+    [self __generatePlaceholder];
   }
 
   return _layout;
 }
 
+- (NSArray<_ASDisplayNodePosition *> *)_filterSublayouts:(NSArray<ASLayout *> *)layouts withIndexes:(NSIndexSet *)indexes
```

Comment:
Selector / Name is not very clear about what this does, although it is fairly readable (nice type annotations)